diff --git "a/C:\\Users\\V1DF1~1.DUB\\AppData\\Local\\Temp\\Htt14ED.tmp\\HttpExceptionMapper-HEAD-left.php" "b/D:\\projects\\MetaPlayer\\meta-player\\library\\Ding\\src\\mg\\Ding\\MVC\\Http\\HttpExceptionMapper.php"
index b26a56c..f0755b3 100644
--- "a/C:\\Users\\V1DF1~1.DUB\\AppData\\Local\\Temp\\Htt14ED.tmp\\HttpExceptionMapper-HEAD-left.php"
+++ "b/D:\\projects\\MetaPlayer\\meta-player\\library\\Ding\\src\\mg\\Ding\\MVC\\Http\\HttpExceptionMapper.php"
@@ -100,7 +100,7 @@ class HttpExceptionMapper implements IMapper
             if (!($exception instanceof $controllerException)) {
                 continue;
             }
-            return array($controller, str_replace('\\', '_', $controllerException) . 'Exception');
+            return array($controller, str_replace('\\', '_', lcfirst($controllerException)) . 'Action');
         }
         return false;
     }
