diff --git "a/C:\\Users\\V1DF1~1.DUB\\AppData\\Local\\Temp\\Htt575A.tmp\\HttpFrontController-HEAD-left.php" "b/D:\\projects\\MetaPlayer\\meta-player\\library\\Ding\\src\\mg\\Ding\\MVC\\Http\\HttpFrontController.php"
index 9b50228..a6526c4 100644
--- "a/C:\\Users\\V1DF1~1.DUB\\AppData\\Local\\Temp\\Htt575A.tmp\\HttpFrontController-HEAD-left.php"
+++ "b/D:\\projects\\MetaPlayer\\meta-player\\library\\Ding\\src\\mg\\Ding\\MVC\\Http\\HttpFrontController.php"
@@ -135,21 +135,24 @@ class HttpFrontController
                 throw new MVCException($url . ' is not a base url.');
             }
             $url = substr($url, $baseUrlLen);
-            $variables = array();
-            if ($method == 'get') {
-                $argsStart = strpos($url, '?');
-                if ($argsStart != false) {
-                    $urlArgs = substr($url, $argsStart + 1);
-                    $arguments = explode('&', $urlArgs);
-                    $variables = array();
-                    foreach ($arguments as $argument) {
-                        $data = explode('=', $argument);
-                        $variables[$data[0]] = isset($data[1]) ? $data[1] : '';
-                    }
-                }
-            } else if ($method == 'post') {
-                $variables = $_POST;
-            }
+            
+            
+            $variables = $_REQUEST;
+            // Val.D.: there is parse_url, but request contains everything what we need
+//            if ($method == 'get') {
+//                $argsStart = strpos($url, '?');
+//                if ($argsStart != false) {
+//                    $urlArgs = substr($url, $argsStart + 1);
+//                    $arguments = explode('&', $urlArgs);
+//                    $variables = array();
+//                    foreach ($arguments as $argument) {
+//                        $data = explode('=', $argument);
+//                        $variables[$data[0]] = isset($data[1]) ? $data[1] : '';
+//                    }
+//                }
+//            } else if ($method == 'post') {
+//                $variables = $_POST;
+//            }
             $action = new HttpAction($url, $variables);
             $action->setMethod($method);
             $mapper = $container->getBean('HttpUrlMapper');
